pkg/controller/controller.go

```diff
@@ -216,6 +222,14 @@ func (ctrl *CSIAttachController) syncPV() {
 		ctrl.pvQueue.AddRateLimited(pvName)
 		return
 	}
+
+	if !ctrl.shouldProvision(pv) {
```

Comment:
IMO it would be worth adding this check into `pvAdded` / `pvUpdated` event handlers, so we skip whole workqueue processing.